kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -139,6 +150,17 @@ private CompletionStage<ProduceRequestData> maybeEncodeProduce(ProduceRequestDat
                 });
     }
 
+    private void generateMetrics(String virtualClusterName, Set<String> unresolvedTopicNames, Map<String, TopicProduceData> topicNameToData) {
+        unresolvedTopicNames.forEach(unresolvedTopic -> {
+            var unEncyptedRecordsTotal = RecordEncryptionMetrics.recordEncryptionPlainRecordsCounter(virtualClusterName, unresolvedTopic).withTags();
+            TopicProduceData data = topicNameToData.get(unresolvedTopic);
```

Comment:
move the metric lookup into the loop.
you can then use unresolvedTopicNames.map(... /* to TopicProduceData */).filter(Objects::notNull).forEach(...)

Reply:
Sorry Keith, I tried this but wasn't able to get the desired result